gn_build.sh

```diff
@@ -99,6 +99,25 @@ else
     echo "Hint: Set \$EFR32_SDK_ROOT to enable building for EFR32"
 fi
 
+opts=$(getopt --longoptions "pincode:,discriminator:" -n "$0" --options "" -- "$@")
```

Comment:
Disagree with creating new bespoke options here, instead we could just generically accept args:

`gn_build.sh chip_device_config_use_test_setup_pin_code=12345`

But really, if you want a custom configuration, I'd actually recommend not using gn_build.sh at all and following the documented instructions for creating a custom configuration.

gn_build.sh is more intended to build what CI builds, without any local deviations.